### DISCUSSION TITLE
Create a new record from the Backend

### DISCUSSION THREAD

**Discussion by LouvAndTech** [pocketbase/pocketbase#1230] [Q&A] [answered]
Hello, 
I have that may seems trivial, but I didn't found how to add a new record on a table from the backend.

My goal is to update periodically my local storage from the githubAPi.
I've successfully retrieve the data from the API but when I use this bit of code to add it to the local DB :
 ```golang
//Insert the new repo
	app.DB().Insert("github_projects", dbx.Params{
		"repo_name":    newRepo.Name,
		"link_to_repo": newRepo.Url,
		"username":     newRepo.Owner,
		"image_link":   newRepo.AvatarUrl,
		"tags":         "",
		"description":  newRepo.Description,
		"stars":        newRepo.Stars,
		"fork":         newRepo.Forks,
		"contributor":  0,
	}).Execute()
```

I end up having a bunch of problematic entry in my db :
[Image here](https://ibb.co/ScrCvsb)

I can see them, but I cannot interact with them and when I tried to edit on of those, it let me edit but once I chose to save change it create a new entry, the difference been that this new entry as an id and work perfectly fine.

Thanks for the help

**Comment by ganigeorgiev** [comment, marked as answer]
`Insert()` performs a plain SQL "INSERT" query with your `dbx.Params` but the id also need to be set if you want to properly insert the data.

How did you create the `github_projects` table and collection, from the Admin UI or via manual `CREATE TABLE` query? I'm not sure how you were even able to insert multiple records with the same empty id. By default all records tables are created with `id TEXT PRIMARY KEY` and the expected behavior would be to fail after the first record with empty id. (**edit**: _The inserted records are with `NULL` id and doesn't trigger the UNIQUE constraint, I'll consider updating the default tables definition and adding an extra `NOT NULL` to the records table `PRIMARY KEY`_).

In any case, you can try deleting the empty id entries by executing something like:
```sql
sqlite3 pb_data/data.db 'delete from github_projects where id is null'
```

With that said I don't recommend directly calling `Insert` unless you really want the raw/plain query behavior. Additionally, please keep in mind that `OnModel*` hook events are fired only when using the `Dao` and a proper `models.Model` instance. Please take a look at https://pocketbase.io/docs/record-methods/#create-new-record.

**Comment by LouvAndTech** [comment]
Thanks a lot !
And thanks for your patience with me.

You solve my problem, I've definitely miss this part of the documentation. 
In my defense, the name of the function as change from 

```golang
record.Set("title", "Lorem ipsum")
```
to 
```golang
record.SetDataValue("title", "Lorem ipsum")
```

But I should definitely have seen that before. 
Thanks again ! 
And thanks for developing PocketBase

**Reply by ganigeorgiev** [reply]
Hm, I guess you are probably using v0.7.x. If you find time please consider migrating to v0.8+ since it includes a lot of improvements and changes to how internally the data is managed.

For a v0.7.x -> v0.8.x migration guide you can refer to https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v080.

After migrating to v0.8 you can safely update your go.mod to the latest [v0.9.2 Release](https://github.com/pocketbase/pocketbase/releases) (_note that there are some minor internal changes to v0.9 too in https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v090_).

**Reply by LouvAndTech** [reply]
How ok thanks again !
